jiant/utils/data_loaders.py

```diff
@@ -286,17 +283,15 @@ def get_tag_list(tag_vocab):
     return tag_list
 
 
-def process_sentence(tokenizer_name, sent, max_seq_len):
-    """process a sentence """
+def truncate_and_tokenize(tokenizer_name, sent, max_seq_len):
+    """Truncate and tokenize a sentence or paragraph."""
     max_seq_len -= 2
-    assert max_seq_len > 0, "Max sequence length should be at least 2!"
+    # TODO: This fn is called potentially millions of times per run -- can we move the assert elsewhere?
+    assert max_seq_len > 0, "Max sequence length should be at least 2 to allow for boundary tokens!"
     tokenizer = get_tokenizer(tokenizer_name)
-    if tokenizer_name.startswith("bert-"):
-        sos_tok, eos_tok = BERT_CLS_TOK, BERT_SEP_TOK
-    else:
-        sos_tok, eos_tok = SOS_TOK, EOS_TOK
+
     if isinstance(sent, str):
-        return [sos_tok] + tokenizer.tokenize(sent)[:max_seq_len] + [eos_tok]
```

Comment:
Note: This change and the next one are big, and will likely mess up most custom tasks. I _tried_ to fix checked-in tasks above.

Reply:
Easy/dumb sanity check: ctrl-f for `[1:]` or `[:-1]` in `tasks.py`? That shouldn't happen anymore, if I'm understanding the code correctly.